src/Data/Avro/Schema.hs

```diff
@@ -230,27 +231,35 @@ instance ToJSON Type where
       Map tn   -> object [ "type" .= ("map" :: Text), "values" .= tn ]
       NamedType (TN tn) -> A.String tn
       Record {..} ->
-        object [ "type"      .= ("record" :: Text)
+        let opts = catMaybes
+               [ ("order" .=)     <$> order
```

Comment:
Don't add optional fields if they have no values.
The reason is that `null` is not a correct value from the `Avro` point of view.
For example, `order` field can only be missing (which means `accending`) or have a value of `accending|descending|ignored`. Having `null` in this field is not acceptable.